src/main/java/hw33/component/ProductRepository.java

```diff
@@ -0,0 +1,33 @@
+package hw33.component;
+
+import jakarta.annotation.PostConstruct;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+@Component
+public class ProductRepository {
+    private List<Product> products;
```

Comment:
Your goal was to use the repository in the Cart
Now you have not related objects